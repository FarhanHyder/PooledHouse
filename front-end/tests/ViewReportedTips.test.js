import React from 'react';
import renderer from 'react-test-renderer';
import Report from '../src/Components/ViewTipInfo/ViewReportedTips';

test('Report should render correct for single element', () => {
  const thisProps = {
    business: 'Poolinos',
    locations: {
      '282 Bowery, New York, NY 10012': {
        'tipsPerHour': 12.8888,
      }
    },
    avgByDay: {
      Sunday: {
        'tipsPerHour': 12.8888
      },
      Monday: {
        'tipsPerHour': 12.8888
      },
      Tuesday: {
        'tipsPerHour': 12.8888
      },
      Wednesday: {
        'tipsPerHour': 12.8888
      },
      Thursday: {
        'tipsPerHour': 12.8888
      },
      Friday: {
        'tipsPerHour': 12.8888
      },
      Saturday: {
        'tipsPerHour': 12.8888
      }
    },
    avgByPosition: {
      Bartender: {
        tipsPerHour: 12.8888
      },
      Server: {
        tipsPerHour: 12.8888
      },
      Barback: {
        tipsPerHour: 12.8888
      },
      Busser: {
        tipsPerHour: 12.8888
      },
      Other: {
        tipsPerHour: 12.8888
      }
    },
    avgByPosShift: {
      PM: {
        Bartender: {
          tipsPerHour: 12.8888
        },
        Server: {
          tipsPerHour: 12.8888
        },
        Barback: {
          tipsPerHour: 12.8888
        },
        Busser: {
          tipsPerHour: 12.8888
        },
        Other: {
          tipsPerHour: 12.8888
        }
      },
      AM: {
        Bartender: {
          tipsPerHour: 12.8888
        },
        Server: {
          tipsPerHour: 12.8888
        },
        Barback: {
          tipsPerHour: 12.8888
        },
        Busser: {
          tipsPerHour: 12.8888
        },
        Other: {
          tipsPerHour: 12.8888
        }
      }
    }
  }

  const component = renderer.create(
    <Report business='Poolinos' locations={thisProps.locations} avgByDay={thisProps.avgByDay}
      avgByPosition={thisProps.avgByPosition} avgByPosShift={thisProps.avgByPosShift}/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
