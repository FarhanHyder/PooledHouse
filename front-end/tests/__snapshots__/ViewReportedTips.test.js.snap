// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report should render correct for single element 1`] = `
<div
  className="container"
>
  <div
    className="card"
  >
    <br />
    <div
      className="row btn-group"
    >
      <button
        className="btn btn-dark"
        onClick={[Function]}
        type="button"
      >
        Tips By Position
      </button>
      <button
        className="btn btn-dark"
        onClick={[Function]}
        type="button"
      >
        Tips By Day
      </button>
      <button
        className="btn btn-dark"
        onClick={[Function]}
        type="button"
      >
        Tips By Shift And Position
      </button>
    </div>
    <br />
  </div>
  <div
    className="row fluid"
  >
    <div
      className="col-4"
    >
      <div
        className="table-responsive-sm"
      >
        <table
          className="table table-dark table-striped table-bordered table-hover"
        >
          <thead
            className="bg-secondary"
          >
            <tr>
              <th>
                Position
              </th>
              <th>
                Hourly Tips
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Bartender
              </td>
              <td>
                $
                12.89
              </td>
            </tr>
            <tr>
              <td>
                Server
              </td>
              <td>
                $
                12.89
              </td>
            </tr>
            <tr>
              <td>
                Barback
              </td>
              <td>
                $
                12.89
              </td>
            </tr>
            <tr>
              <td>
                Busser
              </td>
              <td>
                $
                12.89
              </td>
            </tr>
            <tr>
              <td>
                Other
              </td>
              <td>
                $
                12.89
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      className="col-8"
    >
      <div
        className="card"
      >
        <h3
          className="card-title bg-secondary"
        >
          Average Tips By Position
        </h3>
      </div>
    </div>
  </div>
</div>
`;
